File: gitlfs-client/src/main/java/ru/bozaro/gitlfs/client/HttpExecutor.java
package ru.bozaro.gitlfs.client;

import org.apache.http.client.methods.CloseableHttpResponse;
import org.apache.http.client.methods.HttpUriRequest;

import javax.annotation.Nonnull;
import java.io.Closeable;
import java.io.IOException;

/**
 * Abstract class for HTTP connection execution.
 *
 * @author Artem V. Navrotskiy
 */
public interface HttpExecutor extends Closeable {
  @Nonnull
  CloseableHttpResponse executeMethod(@Nonnull HttpUriRequest request) throws IOException;
}

File: gitlfs-client/src/main/kotlin/ru/bozaro/gitlfs/client/HttpExecutor.kt
package ru.bozaro.gitlfs.client

import org.apache.http.client.methods.CloseableHttpResponse
import org.apache.http.client.methods.HttpUriRequest
import java.io.Closeable
import java.io.IOException

/**
 * Abstract class for HTTP connection execution.
 *
 * @author Artem V. Navrotskiy
 */
interface HttpExecutor : Closeable {
    @Throws(IOException::class)
    fun executeMethod(request: HttpUriRequest): CloseableHttpResponse
}
